src/pkg/artifact/manager.go

```diff
@@ -0,0 +1,112 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package artifact
+
+import (
+	"context"
+	"github.com/goharbor/harbor/src/common/models"
+	"github.com/goharbor/harbor/src/pkg/q"
+	"time"
+)
+
+var (
+	// Mgr is a global artifact manager instance
+	Mgr = NewManager()
+)
+
+// Manager is the only interface of artifact module to provide the
+// management functions for artifacts
+type Manager interface {
+	// List artifacts according to the query
+	List(ctx context.Context, query ...*q.Query) (total int64, artifacts []*Artifact, err error)
```

Comment:
IMO, suggest using `query  *q.Query` to replace `query ...*q.Query` and append more comments to indicate that query can be nil (that means no query parameters)

Reply:
Any way to query out all the useless artifacts (no tags attached)?

Reply:
We can support that by adding a query keyword such as `without-tag=true` if we have the requirement